integration/auth_oidc_test.go

```diff
@@ -36,6 +37,181 @@ type AuthOIDCScenario struct {
 	mockOIDC *dockertest.Resource
 }
 
+func TestOIDCUsernameGrant(t *testing.T) {
+	IntegrationSkip(t)
+	t.Parallel()
+
+	baseScenario, err := NewScenario()
+	if err != nil {
+		t.Errorf("failed to create scenario: %s", err)
+	}
+
+	scenario := AuthOIDCScenario{
+		Scenario: baseScenario,
+	}
+
+	spec := map[string]int{
+		"user1": len(TailscaleVersions),
+	}
+
+	users := make([]string, len(TailscaleVersions))
+	for i := range users {
+		users[i] = "test-user <test-email@example.com>"
+	}
+	user_str := strings.Join(users, ", ")
+
+	oidcConfig, err := scenario.runMockOIDC(defaultAccessTTL, user_str)
+	if err != nil {
+		t.Errorf("failed to run mock OIDC server: %s", err)
+	}
+
+	oidcMap := map[string]string{
+		"HEADSCALE_OIDC_ISSUER":             oidcConfig.Issuer,
+		"HEADSCALE_OIDC_CLIENT_ID":          oidcConfig.ClientID,
+		"CREDENTIALS_DIRECTORY_TEST":        "/tmp",
```

Comment:
Is this for Headscale or mockoidc? if its headscale, then prefix would be good.

Reply:
Are you referring to `CREDENTIALS_DIRECTORY_TEST`? I copied this environment map from the TestOIDCAuthenticationPingAll test. For whatever historical reason, that test passes in the client secret with a file instead of directly, which is the reason for the directory. I could simplify if needed (I don't see why `CREDENTIALS_DIRECTORY_TEST` is a separate env variable).